### PR TITLE
batches: Fix hello world example

### PR DESCRIPTION
**Pull request by eseliger**
This didn't work when there were any file paths that contain spaces. I ran this across 10k workspaces on k8s and it didn't fail - good enough validation?
Big disclaimer: Not a bash expert, I barely understand WHY this fixes it, but definitely don't if this has bad side effects.

## Test plan

See above, ran on k8s.

## App preview:

- [Web](https://sg-web-es-fix-hello-world.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-pelplwmnpf.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Review by BolajiOlajide — APPROVED** [review]
I'm not a bash expert also but I'm surprised this fixes it. I know `IFS` is used to determine how the shell does word splitting but I'm wondering at what point the splitting occurs (if that ever happens) when we mount .

**Comment by eseliger** [comment]
It changes argument splitting to only trigger on newlines, so the output of `find` is split correctly by`\n` but not by spaces in the paths.

**Review by Piszmog — APPROVED** [review]

**Review by courier-new — COMMENTED** [review]
Equally unsure why this fixes things. 🙃 And don't get me wrong, I'm glad it does, but it feels sorta bad that we have to complicate our hello world example with this. We ideally want this example to be as readable and straightforward as possible, and adding this definitely takes away from that to fix a problem that maybe affects <1% of users actually running this spec. 😅

This also reminds me that I totally dropped the ball on reviewing https://github.com/sourcegraph/sourcegraph/pull/33694 that Kelvin from CS opened a while back. It's a different fix for a different problem they were experiencing, but do you know _what_ doesn't work when there are file paths with spaces? Like on the off chance that the problem is with `tee`, would replacing it with `sed` or something else work instead?

Basically, I'm wondering if there's another solution we could use instead maybe. 🙈

**Comment by eseliger** [comment]
> This also reminds me that I totally dropped the ball on reviewing https://github.com/sourcegraph/sourcegraph/pull/33694 that Kelvin from CS opened a while back. It's a different fix for a different problem they were experiencing, but do you know what doesn't work when there are file paths with spaces? Like on the off chance that the problem is with tee, would replacing it with sed or something else work instead?

Yes, the problem is bash, not tee. 
When bash invokes a program, it needs to do something like what we do in go:
```
cmd := newCommand(lookupPath(cmdName))
cmd.Env = getBashEnv()
args := getBashArgs()
cmd.Argc = len(args)
cmd.Argv = args
cmd.Run()
```

The `getBashArgs` function does about this:
```
func getBashArgs(cmdInput string) []string {
split := strings.SplitByAny(cmdInput, getEnv("IFS")) // where IFS defaults to space, newline, tab iirc
return split[1:]
}
```

So it splits the given input string by the IFS. By default, that's strings and newlines. Newlines is what we need, because `find` returns all paths separated by newlines. Space is what we don't want, because it's not a separator for the `find` output. That means that `cmd/the awesome tool/main.go` is turned into `argv := []string{"cmd/the", "awesome", "tool/main.go"}`, of which none exist. Settings IFS explicitly to newline only makes it work, because then we properly respect the splitting character that `find` uses. 

If there's a nicer-looking solution I don't know, this is where my bash knowledge ends :( Maybe the example should just use `search_result_paths` instead..

Hope this helps, and happy to hear opinions :)

**Comment by courier-new** [comment]
Thanks for the breakdown!! That makes more sense. I kinda like the idea of showing off `search_result_paths` (teaching a bit about template variables in the process) for the example instead, if only for the purposes of skirting around this issue. 🙃 But interested to hear what others think.

**Review by LawnGnome — APPROVED** [review]
I'm a little surprised that works, since Alpine `sh` isn't `bash` and `$'\n'` is a `bash`-ism, but given that it apparently does, :+1:.
